lib/ACL/ACLRuleCache.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace OCA\GroupFolders\ACL;
+
+use OCA\GroupFolders\ACL\UserMapping\IUserMapping;
+use OCA\GroupFolders\ACL\UserMapping\IUserMappingManager;
+use OCA\GroupFolders\ACL\UserMapping\UserMapping;
+use OCP\ICache;
+use OCP\ICacheFactory;
+use OCP\IUser;
+
+class ACLRuleCache
+{
+	const DERIVATION = 0.1;
+
+	const DEFAULT_TTL = 3600 * 24;
+
+	const INDEX_TTL = (3600 * 24) + 3600;
+
+	/**
+	 * @var ICache
+	 */
+	private $ruleCache;
+
+	/**
+	 * @var IUserMappingManager
+	 */
+	protected $userMappingManager;
+
+	public function __construct(ICacheFactory $cacheFactory, IUserMappingManager $userMappingManager) {
+		$this->userMappingManager = $userMappingManager;
+		$this->ruleCache = $cacheFactory->createDistributed('acl');
```

Comment:
```suggestion
		$this->ruleCache = $cacheFactory->createDistributed('groupfolders-acl');
```

Maybe a bit safer in terms of naming collisions.